js/data.js

```diff
@@ -6,3 +6,43 @@ var data = {
   editing: null,
   nextEntryId: 1
 };
+
+var $photoUrl = document.querySelector('#imgurl');
+var $img = document.querySelector('img');
+var $form = document.querySelector('form');
+var $title = document.querySelector('#title');
+var $notes = document.querySelector('#notes');
+
+$photoUrl.addEventListener('input', function (event) {
+  if ($photoUrl.value === '') {
+    $img.src = 'images/placeholder-image-square.jpg';
+  } else {
+    $img.src = $photoUrl.value;
+  }
+});
+
+$form.addEventListener('submit', function (event) {
+  event.preventDefault();
+  var titleValue = $title.value;
+  var photoUrlValue = $photoUrl.value;
+  var notesValue = $notes.value;
+
+  var entry = {
+    title: titleValue,
+    photoUrl: photoUrlValue,
+    notes: notesValue,
+    entryId: data.nextEntryId
+  };
+  data.nextEntryId++;
+  data.entries.unshift(entry);
+});
```

Comment:
All of this functionality should be in the `main.js` file. The only thing you want in the data.js file is the `data` object and the local storage functionality.